src/pages/user/viewer/details.jsx

```diff
@@ -1,36 +1,102 @@
+import { faStar as faStarSolid, faCheck, faCode, faFileContract, faMapPin, faMars, faPhone, faPhoneSlash, faPrint, faUserPen, faVenus } from '@fortawesome/free-solid-svg-icons';
+import { Colors } from '../../../theme';
+import { dateOfBirthToAge } from '../../../utils/user';
+import { AuthenticationContext } from '../../../components/authentication';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
 
 const S = {
     Avatar: styled.img`
         width: 256px;
-        border: 1px solid rgb(235, 235, 235);
+        border: 1px solid ${Colors.Gray200};
+        margin-bottom: 1em;
+
+        @media screen and (max-width: 480px) {
+            width: 100%;
+        }
     `,
 }
 
-export const UserViewerDetails = ({ user }) => {
+export const UserViewerDetails = ({ inheritUser, reloadFunction }) => {
+
+    const [ user, setUser ] = useState(inheritUser);
+    
+    let history = useHistory();
+
+    // Import the following React contexts:
+    const authorizedUser = useContext(AuthenticationContext);
+
+    const [ activationStateButtonAvailibility, setActivationButtonAvailibility ] = useState(false);
+    const [ modifyUserStateButtonAvailibility, setModifyUserButtonAvailibility ] = useState(false);
+    const [ deleteAvatarButtonAvailibility, setDeleteAvatarButtonAvailibility ] = useState(false);
+
     const [ membershipState, setMembershipState ] = useState(null);
-    const [ activationState, setActivationState] = useState(null);
-    const [ loading, setLoading ] = useState(false);
+    const [ activationState, setActivationState ] = useState(null);
+
+    const [ loading, setLoading ] = useState(true);
 
     const reload = async () => {
         setLoading(true);
-        const [ membershipState, activationState ] = await Promise.all([
-            User.getUserMembershipStatus(user.uuid),
-            User.getUserActivationState(user.uuid),
-        ])
-        setMembershipState(membershipState);
-        setActivationState(activationState);
+
+        setUser(await User.getUser(inheritUser.uuid));
+
+        // Check if user has "admin" role and make the following functions available:
+        if (authorizedUser.roles.includes("admin")) {
+            setActivationButtonAvailibility(true);
+            setModifyUserButtonAvailibility(true);
+        }
+
+        // Avatar button availability logic, check if the user is him/herself or is admin or hr_admin, and check if the user has an avatar to make the button available:
+        if (user.avatar_uuid) {
+            if((authorizedUser.roles.includes("admin") || authorizedUser.roles.includes("hr_admin")) || authorizedUser.authUser.uuid == user.uuid) {
+                setDeleteAvatarButtonAvailibility(true);
+            }
+        }
+
+        // Try to get user information and set the information as states which can be used later or throw an error.
+        try {
+            setActivationState(await User.getUserActivationState(user.uuid));
```

Comment:
You should do this from the Promise.all above to speed up loading

Reply:
Fixed(?)